packages/cli/src/zosfiles/edit/Edit.utils.ts

```diff
@@ -0,0 +1,225 @@
+                response = await Upload.fileToDataset(session, lfFile.dir, fileName, {etag: lfFile.zosResp.apiResponse.etag});
+            }
+            if (response.success){
+                // If matching etag & successful upload, destroy tmp file -> END
+                await this.destroyTempFile(lfFile.dir);
+                return true;
+            }
+        }catch(err){
+            const etagMismatchCode = 412;
+            if (err.errorCode && err.errorCode == etagMismatchCode){
+                // open a fileComparision
+                await this.fileComparison(session, commandParameters);
+                //alert user that the version of document they've been editing has changed
+                //ask if they want to continue working with their stash (local file)
+                const continueToUpload: boolean = await this.promptUser(Prompt.continueToUpload, lfFile.dir);
+                if (continueToUpload){
+                    // Refresh the etag of lfFile (keep stash)
+                    if (commandParameters.positionals.includes('uss')){
+                        lfFile.zosResp = await Download.ussFile(session, commandParameters.arguments.file, {returnEtag: true, file: tmpdir()+'toDelete'});
+                    }
+                    else{
+                        lfFile.zosResp = await Download.dataSet(session, fileName,
+                            {returnEtag: true, file: tmpdir()+'toDelete'});
+                    }
+                    this.destroyTempFile((tmpdir()+'toDelete'));
+                    // upload lf version to mf
+                    return false;
+                }else{
+                    // keep editing lf
+                    if (commandParameters.positionals.includes('uss')){
+                        lfFile.zosResp = await Download.ussFile(session, commandParameters.arguments.file, {returnEtag: true, file: lfFile.dir});
+                    }
+                    else{
+                        lfFile.zosResp = await Download.dataSet(session, fileName,
+                            {returnEtag: true, file: lfFile.dir});
+                    }
+                    // open lf in editor
+                    await this.makeEdits(commandParameters);
+                    return false;
+                }
+            }
+            throw new ImperativeError({
+                msg: TextUtils.chalk.red(`Command terminated. Stashed file will persist: ${lfFile.dir}`),
+                causeErrors: err
+            });
+        }
+    }
+
+    public static async destroyTempFile(tmpDir:string): Promise<void>{
+        unlink (tmpDir, (err) => {
+            if (err) throw new ImperativeError({
+                msg: `Temporary file could not be deleted: ${tmpDir}`,
+                causeErrors: err
+            });
+        });
+    }
+
+    public static async errorHandler(error: RestClientError): Promise<void>{
+        if (error.causeErrors.code == 'ENOTFOUND'){
+            throw new ImperativeError({
```

Comment:
Decided to make a function that both USS and DS handlers can use so error description is only written once. This makes errors pretty long whereas when I would through it in the catch of the handler it was quite short. Unsure if i should keep this method or just throw errors back in the handlers and have some repeated code. What do you guys think?

Longer error from uss/ds shared method:
![image](https://user-images.githubusercontent.com/112635587/230136518-31d62d3f-5e7a-4a8f-a86e-1e70b3b253fa.png)

Concise error and repeated code:
![image](https://user-images.githubusercontent.com/112635587/230137361-c7763d99-1b2f-4423-9c86-53b343db0aaf.png)

Reply:
I think that the longer errors (and their stack trace) make it appear as if the command crashed, instead of its actual behavior in which it detected some problem, reported the problem, and then gracefully ended.